src/components/NcCheckboxRadioSwitch/NcCheckboxRadioSwitch.vue

```diff
@@ -492,49 +540,83 @@ $spacing: 4px;
 
 	&--disabled &__label {
 		opacity: $opacity_disabled;
-		.checkbox-radio-switch__icon {
+		.checkbox-radio-switch__icon > * {
 			color: var(--color-main-text)
 		}
 	}
 
-	&:not(&--disabled) &__label:hover,
-	&:not(&--disabled) &__label:focus-within {
-		background-color: var(--color-primary-light);
+	&:not(&--disabled, &--checked):focus-within &__label,
+	&:not(&--disabled, &--checked) &__label:hover {
+		background-color: var(--color-background-hover);
+	}
+
+	&--checked:not(&--disabled):focus-within &__label,
+	&--checked:not(&--disabled) &__label:hover {
+		background-color: var(--color-primary-light-hover);
 	}
 
 	// Switch specific rules
-	&-switch:not(&--checked) &__icon {
+	&-switch:not(&--checked) &__icon > * {
 		color: var(--color-text-maxcontrast);
 	}
 
 	// If switch is checked AND disabled, use the fade primary colour
-	&-switch.checkbox-radio-switch--disabled.checkbox-radio-switch--checked &__icon {
+	&-switch.checkbox-radio-switch--disabled.checkbox-radio-switch--checked &__icon > * {
 		color: var(--color-primary-element-light);
 	}
 
+	&--button-variant {
+		--border-radius-nc-radio-switch: var(--border-radius-large);
+
+		&-h-grouped {
+			--border-radius-nc-radio-switch: var(--border-radius-pill);
```

Comment:
Here I would say to use `calc(var(--default-clickable-area) / 2)` instead.

This will result in a consistent border-radius of 22px.
- Then it looks pill-style for single-line switches like in Forms.
- And for double-line (like with icon above text like in the Files sidebar) it will also look nice and not have a huge radius


@susnux Hope that clears it up, sorry for the confusion. :)

Reply:
Ah, just saw that @raimund-schluessler already commented the same – that's exactly what I meant, thanks! :)

Reply:
> Then it looks pill-style for single-line switches like in Forms.

BTW we use `--border-radius-pill` on forms too, so this might look a bit different when using this.